internal/registerehrid/register-ehrid.go

```diff
@@ -88,19 +90,16 @@ func RegisterEhrid(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func register(ctx context.Context, registration structs.Registration) (string, error) {
+func register(ctx context.Context, store store.Storer, ehrid string, registration structs.Registration) (string, error) {
 	logger := logging.FromContext(ctx)
 
-	var ehrid string
-
 	err := retry.Do(
 		func() error {
-			ehrid = utils.GenerateEHrid()
-			var doc = firebase.FirestoreClient.Collection(constants.CollectionRegistrations).Doc(ehrid)
+			var doc = store.Doc(ehrid)
```

Comment:
Nope. The change is invalid - ehrid must be retried because there might be a collision with existing.

Reply:
👍  how about now?

Reply:
Much better. Now there are only necessary things changed.  
I would prefer to remove the empty line at the start of the lambda, but... :innocent:

Reply:
done!